.github/workflows/golangci-lint.yaml

```diff
@@ -0,0 +1,49 @@
+name: golangci-lint
+on:
+  pull_request:
+
+jobs:
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-18.04
+    steps:
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v2
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.41
+
+          # Optional: working directory, useful for monorepos
+          # working-directory: somedir
+
+          # Optional: golangci-lint command line arguments.
+          args: -E gosec  --timeout=6m
+
+          # Optional: show only new issues if it's a pull request. The default value is `false`.
+          # only-new-issues: true
+
+  get-runner-shell:
```

Comment:
as this workflow is for golangci-lint do we need to shell access?

Reply:
oh I was just commenting about the requirement of this changes...commented below

Reply:
if you want I can spilt these changes in two files

Reply:
that seems reasonable.

Reply:
done